processor/transformprocessor/internal/common/processor.go

```diff
@@ -0,0 +1,415 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottldatapoints"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottllogs"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlresource"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlscope"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottltraces"
+)
+
+type Context interface {
+	// IsContext dummy method for type safety
+	IsContext()
+}
+
+type TracesContext interface {
+	ProcessTraces(td ptrace.Traces) error
+}
+
+type MetricsContext interface {
+	ProcessMetrics(td pmetric.Metrics) error
+}
+
+type LogsContext interface {
+	ProcessLogs(td plog.Logs) error
+}
+
+var _ Context = &ResourceStatements{}
+var _ TracesContext = &ResourceStatements{}
+var _ MetricsContext = &ResourceStatements{}
+var _ LogsContext = &ResourceStatements{}
+
+type ResourceStatements struct {
```

Comment:
This can be in the ottl package?

Reply:
Ya I am pretty sure it should live there eventually, but since that is a public package and this might change lot in the coming weeks I wanted to flush it out in the transform processor to start and then move it over once its ready to be widely used.